plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -124,7 +133,27 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
             if (summary.getColumnSizes() != null) {
                 Long columnSize = summary.getColumnSizes().get(fieldId);
                 if (columnSize != null) {
-                    columnBuilder.setDataSize(Estimate.of(columnSize));
+                    if (columnHandle.getBaseType() instanceof FixedWidthType) {
+                        // columnSize is the size on disk and Trino column stats is size in memory.
+                        // The relation between the two is type and data dependent.
+                        // However, Trino currently does not use data size statistics for fixed-width types
+                        // (it's not needed for them), so do not report it at all, to avoid reporting some bogus value.
+                    }
+                    else {
+                        if (idToType.get(columnHandleTuple.getKey()).typeId() == Type.TypeID.FIXED) {
+                            Types.FixedType fixedType = (Types.FixedType) idToType.get(columnHandleTuple.getKey());
+                            columnSize = (long) fixedType.length();
+                        }
+                        else if (columnHandle.getBaseType() == VARCHAR) {
+                            // Tests showed that for VARCHAR multiplying by 2.7 seems like a good heuristic
+                            columnSize = (long) (columnSize * 2.7);
+                        }
+                        else if (columnHandle.getBaseType() == VARBINARY) {
+                            // Tests showed that for VARCHAR multiplying by 1.4 seems like a good heuristic
+                            columnSize = (long) (columnSize * 1.4);
```

Comment:
Any comment how 1.4 was chosen? (same for 2.7 above)
Can you add a comment?

initially we thought 10x will be OK
are you measuring on some "real data", or using the "random hex" test case we run internally?

also, note that size on disk can be very small when data is dictionary encoded. maybe it's safer to overshoot rather than undershoot? cc @raunaqmorarka @sopel39 for over- vs under-estimation

Reply:
Comment added. 

```
// Tested using item table from TPCDS benchmark
// compared column size of item_desc column stored inside files
// with length of values in that column reported by trino
columnSize = (long) (columnSize * 2.7);
}
else if (columnHandle.getBaseType() == VARBINARY) {
// Tested using VARBINARY columns with random, both in length and content, data
// compared column size stored inside parquet files with length of bytes written to it
columnSize = (long) (columnSize * 1.4);
```

Does it make sense?

Reply:
> // Tested using item table from TPCDS benchmark
> // compared column size of item_desc column stored inside files
> // with length of values in that column reported by trino
> columnSize = (long) (columnSize * 2.7);

this is awesome!



> else if (columnHandle.getBaseType() == VARBINARY) {
> // Tested using VARBINARY columns with random, both in length and content, data
> // compared column size stored inside parquet files with length of bytes written to it
> columnSize = (long) (columnSize * 1.4);

how was the data randomized?

truly random data should not compress at all, so you should have gotten ~1.0 factor

Reply:
I took random strings, change them to byte and save that data, then I compared size on disk with lengths of byte arrays in memory. If you think that we should have 1.0 here then I can do it.

Reply:
Random ascii or radnom full Unicode?

> If you think that we should have 1.0 here then I can do it.

i did not say that. I am asking about methodology.

Reply:
I used `org.apache.commons.lang.RandomStringUtils`. From its javadoc: `Characters will be chosen from the set of alpha-numeric characters as indicated by the arguments`

Reply:
It should be close to 1 for random bytes.
Since you were generating alpha-numeric characters, the byte repertoire was limited, thus data was compressible, and hence > 1.0 factor.

I am fine keeping the 1.4 value, but we need to code-comment that the methodology was not very real-life-like and some better heuristic value could be proposed in the future.

Reply:
But is it better to keep it 1.4 with code comment or to change it to 1 ? What do you think is betteR?

Reply:
Not a big difference, let's go with a bigger number.

The only important part is the code comment -- don't pretend this is some "very smart value"

Reply:
> also, note that size on disk can be very small when data is dictionary encoded. maybe it's safer to overshoot rather than undershoot? cc @raunaqmorarka @sopel39 for over- vs under-estimation

Overshooting is safer.